compiler/qsc_data_structures/src/span.rs

```diff
@@ -16,6 +16,28 @@ pub struct Span {
     pub hi: u32,
 }
 
+impl Span {
+    /// Returns true if the position is within the range.
+    #[must_use]
+    pub fn contains(&self, offset: u32) -> bool {
+        (self.lo..self.hi).contains(&offset)
```

Comment:
Maybe the compiler is smart enough to optimize this away, but I think a comparison (`<` etc) with the `lo` and `hi` offsets is more efficient than comparing against each member of the range.

Reply:
Oh, if you do it manually that actually issues the following clippy lint warning: https://rust-lang.github.io/rust-clippy/rust-1.78.0/index.html#/manual_range_contains. That's why started using `Range::contains`.

Reply:
Just to verify I went to Compiler Explorer and they generate the same assembly even with opt-level=1 (basic optimizations). So I think it's fine.

Reply:
Lints like these are why Clippy drives me crazy. "More readable" according to who?! Come on!

But in any case, following convention is good. Compiler optimizations are good. Thanks for checking!